src/plugins/intel_gna/src/gna_graph_compiler.cpp

```diff
@@ -789,19 +903,18 @@ void GNAGraphCompiler::finalizeConvolution2DPrimitive(InferenceEngine::CNNLayerP
 
     connectOutput(layer, ptr_outputs, num_data_bytes_out);
 
-    const auto kernelHW = convolution._kernel_y * convolution._kernel_x;
-
-    std::vector<uint8_t> transposedWeights;
-    const auto singleKernelSize = in_channels * kernelHW * convolution.precision.size();
-    const auto kernelPad = Gna2RoundUp(singleKernelSize, 16) - singleKernelSize;
-    for (uint32_t k = 0; k < convolution._out_depth; k++) {
-        uint8_t* ptr_filt_current = convolution._weights->cbuffer().as<uint8_t*>() + k * singleKernelSize;
-        auto transposedPart = transposeMatrix(ptr_filt_current, convolution.precision.size(), in_channels, kernelHW);
-        transposedWeights.insert(transposedWeights.end(), transposedPart.begin(), transposedPart.end());
-        transposedWeights.resize(transposedWeights.size() + kernelPad);
-    }
-
-    gnamem->getQueue(REGION_RO)->push_local_ptr(layer, ptr_weights, transposedWeights.data(), transposedWeights.size());
+    Convolution2DWeightsAdaptorData weights_data = {convolution._kernel_x,
```

Comment:
Bit of an overhead for me to create a struct by passing all members here, just to pass the struct as a parameter in the next line.